pkg/auth/dependency/mfa/provider_impl.go

```diff
@@ -35,7 +35,11 @@ func (p *providerImpl) GenerateRecoveryCode(userID string) ([]string, error) {
 }
 
 func (p *providerImpl) ListAuthenticators(userID string) ([]interface{}, error) {
-	return p.store.ListAuthenticators(userID)
+	authenticators, err := p.store.ListAuthenticators(userID)
+	if err != nil {
+		return nil, err
+	}
+	return MaskAuthenticators(authenticators), nil
```

Comment:
Not sure if it's good to mask authenticators at provider layer: in future, there may be some admin APIs would return the authenticator list without masking.

Reply:
In that case, I will remove `ListAuthenticator` and add `ListRawAuthenticator` and `ListMaskedAuthenticator`. For now, there is no use case of listing raw authenticator so I prefer we keep it.